pkg/kubernetes/client/v1alpha1/client.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	env "github.com/openebs/maya/pkg/env/v1alpha1"
+	"github.com/pkg/errors"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"strings"
+)
+
+const (
+	// K8sMasterIPEnvironmentKey is the environment variable key used to
+	// determine the kubernetes master IP address
+	K8sMasterIPEnvironmentKey env.ENVKey = "OPENEBS_IO_K8S_MASTER"
+	// KubeConfigEnvironmentKey is the environment variable key used to
+	// determine the kubernetes config
+	KubeConfigEnvironmentKey env.ENVKey = "OPENEBS_IO_KUBE_CONFIG"
+)
+
+// getInClusterConfigFunc is a typed function to get incluster config
+//
+// NOTE:
+//  functional makes it simple to mock
+type getInClusterConfigFunc func() (*rest.Config, error)
+
+// buildConfigFromFlagsFunc is a typed function to get desired
+// kubernetes config
+//
+// NOTE:
+//  functional makes it simple to mock
+type buildConfigFromFlagsFunc func(string, string) (*rest.Config, error)
+
+// getKubeMasterIPFunc is a typed function to get kubernetes master
+// IP address
+//
+// NOTE:
+//  functional makes it simple to mock
+type getKubeMasterIPFunc func(env.ENVKey) string
+
+// getKubeConfigPathFunc is a typed function to get kubernetes config
+// path
+//
+// NOTE:
+//  functional makes it simple to mock
+type getKubeConfigPathFunc func(env.ENVKey) string
+
+// getKubernetesClientsetFunc is a typed function to get kubernetes
+// clientset
+//
+// NOTE:
+//  functional makes it simple to mock
+type getKubernetesClientsetFunc func(*rest.Config) (*kubernetes.Clientset, error)
+
+// GetConfigFunc is a typed function to get kubernetes config
```

Comment:
Redundant comment -- Line no 43
How these two are different?

Reply:
will address this